CRM/Contribute/BAO/Contribution.php

```diff
@@ -3846,6 +3804,11 @@ public static function completeOrder($input, $recurringContributionID, $contribu
     if (!$contributionID) {
       $contributionResult = self::repeatTransaction($input, $contributionParams);
       $contributionID = $contributionResult['id'];
+      if ($contributionParams['contribution_status_id'] === $completedContributionStatusID) {
+        // Ideally add deprecation notice here & only accept pending for repeattransaction.
+        return self::completeOrder($input, NULL, $contributionID);
+      }
```

Comment:
@mattwire yeah - so instead of doing the rest of the function to complete the order it calls itself - which kinda has the same effect but moves us towards splitting it out so we call `repeatTransaction` instead of `completeOrder` when we want to `repeatTransaction`

The main blocker before was we couldn't return at this point for non-completed as they would miss the pledge handling